text/0268-acceptance-testing-refactor.md

```diff
@@ -0,0 +1,370 @@
+(and implemented in
+[ember-qunit](https://github.com/emberjs/ember-qunit)@3.0.0) will be modified to add the same DOM interaction helpers mentioned above:
+
+* setup importable `click` helper method
+* setup importable `tap` helper method
+* setup importable `triggerKeyEvent` helper method
+* setup importable `triggerEvent` helper method
+* setup importable `fillIn` helper method
+* setup importable `focus` helper method
+* setup importable `blur` helper method
+* setup importable `waitUntil` helper method
+* setup importable `waitFor` helper method
+
+Once implemented, `setupRenderingTest` and `setupAcceptanceTest` will diverge from each other in very few ways.
+
+## Changes from Current System
+
+Here is a brief list of the more important but possibly understated changes
+being proposed here:
+
+* The global test helpers that exist now, will no longer be present (e.g.
+  `click`, `visit`, etc) and instead will be available on the test context as
+  well as importable helpers.
+* `this.owner` will now be present and allow (for the first time 🎉) overriding
+  items in the container/registry.
+* The new system will leverage the `Ember.Application` /
+  `Ember.ApplicationInstance` split so that we can avoid creating an
+  `Ember.Application` instance per-test, and instead leverage the same system
+  that FastBoot itself uses to avoid running initializers for each acceptance
+  test.
+* Implicit promise chaining will no longer be present. If your test needs to
+  wait for a given promise, it should use `await` (which will wait for the
+  system to "settle" in similar semantics to today's `wait()` helper).
+* The test helpers that are included by a new default ember-cli app will be no
+  longer needed and will be removed from the new application blueprint. This
+  includes:
+  * `tests/helpers/resolver.js`
+  * `tests/helpers/start-app.js`
+  * `tests/helpers/destroy-app.js`
+  * `tests/helpers/module-for-acceptance.js`
+
+## Examples
+
+### Test Helper
+
+Assuming the following input:
+
+```js
+import Ember from 'ember';
+
+export function withFeature(app, featureName) {
+  let featuresService = app.__container__.lookup('service:features');
+  featuresService.enable(featureName);
+}
+
+Ember.Test.registerHelper('withFeature', withFeature);
+```
+
+In order for an addon to support both the existing acceptance testing system, and the new system it could replace that helper with the following:
+
+```js
+import { registerAsyncHelper } from '@ember/test';
+
+export function enableFeature(owner, featureName) {
+  let featuresService = this.owner.lookup('service:features');
+  featuresService.enable(featureName);
+}
+
+export default function setupFeatures(hooks, featureName) {
+  hooks.beforeEach(function() {
+    setupFeature(this.owner, featureName);
+  })
+}
+
+registerAsyncHelper('withFeature', function(app, featureName) {
+  setupFeature(app.__container__, featureName);
```

Comment:
I think this is meant to call `enableFeature`

Reply:
DOH!  Fixed (both)!